sender.go

```diff
@@ -60,7 +60,7 @@ func (s *Sender) Send(ctx context.Context, msg *Message) error {
 	case <-s.link.Detached:
 		return s.link.err
 	case <-ctx.Done():
-		return fmt.Errorf("awaiting send: %v", ctx.Err())
+		return fmt.Errorf("awaiting send: %w", ctx.Err())
```

Comment:
The "awaiting send" look like to me that this error can only happen if the timeout occur before sending the message on the link. Looking at the code, if I understand correctly, it may also happen "in transit", right ?